Cleanup & Fixes on xp0-double

kamsar opened the pull request:
I cleaned up a few unused parameters, added `ErrorActionPreference` to make the PowerShells exit on error instead of continue, and added a README with some setup notes specific to xp0-double.

Also removed a CM package param that is not present in the 8.2U3 XP0 package (https://github.com/kamsar/Sitecore-Azure-Scripts/commit/2fe739bcd7ecd37eee78dcc1a0e350b7a12c6ee6)

robhabraken commented on xp0-double/README.txt:
```diff
@@ -0,0 +1,5 @@
+This deploys a XP deployment with a CM and CD (but no reporting or processing instance like XP-full uses)
+
+For this template to work, you need an XP0 CM web deploy package, and an XP1 CD web deploy package for the two web apps. In other words you need aspects of both the XP0 and XP1 packages Sitecore provides.
```
Correct. Very good addition, forget to mention that :-)

robhabraken commented:
Good additions Kam, added some questions; if you could answer those, I'll merge the PR and also update the other sets (single and full).

kamsar commented:
Not seeing any questions? Did you start a review and not complete it perhaps?

robhabraken commented:
that's strange.. sure I placed comments, now I don't see them, will do again

robhabraken commented on xp0-double/xp0d-azuredeploy-msdeploy.json:
```diff
@@ -190,9 +190,7 @@
                   "Application Insights Role": "CM",
                   "KeepAlive Url": "[concat('https://', reference(resourceId('Microsoft.Web/sites', variables('cmWebAppNameTidy'))).hostNames[0], '/sitecore/service/keepalive.aspx')]",
-                  "License Xml": "[variables('licenseXml')]",
-                  "IP Security Client IP": "[parameters('security_clientIp')]",
```
My set isn't U3 yet, did not have time to check U2 vs U3 changes. But I would actually like to use these params, and if you leave the params out of your param file, they will be ignored. So please do not remove them.

kamsar replied:
I only pulled them because the U3 XP0 CM package no longer had that IP Security Client IP Mask setting and it caused errors when it was passed (that the webdeploy package did not define it).

robhabraken replied:
ah, I see, but no, it's a little bit different: the CM package doesn't have it, because I borrowed the SCWDP from the Single (XP0) setup and that one doesn't have IP-security at all (because if you run that as a single instance, you don't want to secure it on IP). However, I tend to use XP0 single and double in acceptance and test environments only, so I _do_ want IP security.

What I did is I added this IP setting on purpose, and to make sure the SCWDP worked with this ARM templates, I added the IP settings to the SCWDP manually. You could just copy the param in the parameters.xml.

kamsar replied:
oic so this template is really designed to run against a customized variant of the xp0 scwdp. Right? (if true, that should get documented)

robhabraken replied:
sorry, yes, as an example, it does.. but I assume one runs them from their own repo or fork, so don't have to pull back any changed settings back into this - all your other additions and remarks are very welcome and I would like to update my demo set with them, but this one especially should either be removed on your side, or added within your SCWDP

robhabraken commented on xp0-double/xp0d-azuredeploy-msdeploy.json:
```diff
@@ -116,18 +116,6 @@
     },
-    "analytics_mongodb_connectionstring": {
-      "type": "securestring"
-    },
```
I'm pretty sure mongo conn strings are used on deployment. You know, msdeploy.json is used for both vanilla and redeploy scripts. So I don't think those should be removed.

kamsar replied:
It's only using the `Tidy` versions (search the file for uses to verify), e.g. 

`"analyticsMongoDbConnStrTidy": "[parameters('provisioningOutput').analyticsMongoDbConnStrTidy.value]",`

It seems to be getting all of this data from the provisioning settings, and it's not erroring with the removal of that property.

robhabraken replied:
your absolutely right, my mistake - I used the provisioning output for this, so my edit made these parameters redundant; can indeed be removed

kamsar commented:
Added the variable back as requested and added a note in the README that a custom package is required or what do to to get it stock-compatible.

(in my case I'm basically using the scwdp just to restore the databases; my ongoing deployments' files are entirely owned by separate scripts).

BTW, the technique we talked about on Slack the other day so far seems to be working fine. About 30 seconds to complete the web deploy/files part of a deployment 😀

robhabraken commented:
Thanks Kam!

Great to hear your technique is working fine and 30 sec is really fast :-)
